core/schema/sdk.go

```diff
@@ -99,165 +86,125 @@ func (s *moduleSchema) builtinSDK(ctx context.Context, sdkName string) (SDK, err
-	sdkMod, err := s.loadModuleTypes(ctx, sdkMod)
+func (s *APIServer) newModuleSDK(ctx context.Context, sdkModMeta *core.Module) (*moduleSDK, error) {
+	sdkMod, err := s.AddModFromMetadata(ctx, sdkModMeta, nil)
 	if err != nil {
-		return nil, fmt.Errorf("failed to load sdk module types: %w", err)
+		return nil, fmt.Errorf("failed to add sdk module to dag: %w", err)
 	}
-
-	return &moduleSDK{moduleSchema: s, mod: sdkMod}, nil
+	return &moduleSDK{APIServer: s, mod: sdkMod}, nil
 }
 
 // Codegen calls the Codegen function on the SDK Module
-func (sdk *moduleSDK) Codegen(ctx context.Context, mod *core.Module) (*core.GeneratedCode, error) {
-	sdkModuleName := gqlObjectName(sdk.mod.Name)
-	var sdkModuleOriginalName string
-	funcName := "Codegen"
-	var codegenFn *core.Function
-	for _, obj := range sdk.mod.Objects {
-		if obj.AsObject.Name == sdkModuleName {
-			sdkModuleOriginalName = obj.AsObject.OriginalName
-			for _, fn := range obj.AsObject.Functions {
-				if fn.Name == gqlFieldName(funcName) {
-					codegenFn = fn
-					break
-				}
-			}
-		}
-	}
-	if codegenFn == nil {
-		return nil, fmt.Errorf("failed to find required Codegen function in SDK module %s", sdkModuleName)
+//
+//nolint:dupl
+func (sdk *moduleSDK) Codegen(ctx context.Context, mod *UserMod) (*core.GeneratedCode, error) {
+	mainModObj, err := sdk.mod.MainModuleObject(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get main module object for sdk module %s: %w", sdk.mod.Name(), err)
 	}
-
-	schemaView, err := sdk.installDeps(ctx, mod)
+	codegenFn, ok, err := mainModObj.FunctionByName(ctx, "Codegen")
 	if err != nil {
-		return nil, fmt.Errorf("failed to install deps during %s module sdk codegen: %w", sdkModuleName, err)
+		return nil, fmt.Errorf("failed to get Codegen function in SDK module %s: %w", sdk.mod.Name(), err)
+	}
+	if !ok {
+		return nil, fmt.Errorf("failed to find required Codegen function in SDK module %s: %w", sdk.mod.Name(), err)
 	}
-	introspectionJSON, err := schemaView.schemaIntrospectionJSON(ctx)
+
+	introspectionJSON, err := mod.SchemaIntrospectionJSON(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get schema introspection json during %s module sdk codegen: %w", sdkModuleName, err)
+		return nil, fmt.Errorf("failed to get schema introspection json during %s module sdk codegen: %w", sdk.mod.Name(), err)
 	}
 
-	srcDirID, err := mod.SourceDirectory.ID()
+	srcDirID, err := mod.metadata.SourceDirectory.ID()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get source directory id: %w", err)
 	}
 
-	result, err := sdk.moduleSchema.functionCall(ctx, codegenFn, functionCallArgs{
-		Module: sdk.mod,
-		Input: []*core.CallInput{
-			{
-				Name:  "modSource",
-				Value: srcDirID,
-			},
-			{
-				Name:  "subPath",
-				Value: mod.SourceDirectorySubpath,
-			},
-			{
-				Name:  "introspectionJson",
-				Value: introspectionJSON,
-			},
+	result, err := codegenFn.Call(ctx, true, nil, nil, []*core.CallInput{
```

Comment:
nit: Hard to tell what these `true, nil, nil` values are - maybe worth a `CallOpts` or something?

Reply:
Agree, fixed